### DISCUSSION TITLE
Allow User To View "Firmware Reports" Prior To Submission

### DISCUSSION THREAD

**Discussion by vindicatorr** [fwupd/fwupd#6344] [Ideas]
I just updated my Arch system and saw new user/group added which got me to look what fwupd was all about and boy was I pleased!

I updated one of my devices which had firmware update from 2017 that I didn't know about.

When I ran `$ fwupdmgr get-updates` again, I saw:
```
Devices that have been updated successfully:
 • <device> (<firmware range>)
Uploading firmware reports helps hardware vendors to quickly identify failing and successful updates on real devices.
Upload report now? (Requires internet connection) [Y|n]: 
```

It'd be nice to include what the report "content" will be.
I figure if it's short, it could just be displayed, or an option to view it (scrollable), or if it's already saved to a file, then mention that file path.

Also, I'd think it'd be better to offer that option at the end of an update, rather than just after a re-check which may not happen after quite some time, in which case the manufacturer would not be aware of success/failure.
But I could foresee an issue if an update failed and the system became unusable until things were restored, then re-displaying the submission option would be good if no submission had been made.

**Comment by hughsie** [comment]
> It'd be nice to include what the report "content" will be.

If you say "yes" there you get to see the JSON.

> Also, I'd think it'd be better to offer that option at the end of an update

We don't know if the update was successfully applied until after the reboot unfortunately.

**Reply by vindicatorr** [reply]
> If you say "yes" there you get to see the JSON.

Oh? Is that _before_ the upload? Because `Upload report now?` is making me think it isn't.

> We don't know if the update was successfully applied until after the reboot unfortunately.

Odd, I got it before rebooting... It was the USB wireless receiver for an input device though.

**Comment by hughsie** [comment]
> Oh? Is that before the upload? Because Upload report now? is making me think it isn't.

Yes, but you're right -- we should probably tell the user that we're going to show them what we upload before we upload. @superm1 how about something like:

    Uploading firmware reports helps hardware vendors to quickly identify failing and successful updates on real devices.
    Review the report now? (Requires internet connection) [Y|n]: 

> Odd, I got it before rebooting

It depends on the device -- some are insta-apply.

**Reply by superm1** [reply]
No opposition

**Reply by hughsie** [reply]
https://github.com/fwupd/fwupd/pull/6354
